app/tasks.py

```diff
@@ -59,7 +59,27 @@ def ocr_task(self, pdf_bytes, strategy_name, pdf_hash, ocr_cache, prompt, model)
             num_chunk += 1
             extracted_text += chunk['response']
 
-    self.update_state(state='DONE', meta={'progress': 100 , 'status': 'Processing done!', 'start_time': start_time, 'elapsed_time': time.time() - start_time})  # Example progress update
+    # Optional call to generate tags and summary
+    if prompt and model:
```

Comment:
Please add an option `generate_metadata` - by default set to true; only when set the metadata is being generated - no material if `prompt` was given or not